src/Umbraco.Core/Logging/DisposableTimer.cs

```diff
@@ -85,19 +118,55 @@ protected override void DisposeResources()
             {
                 if (_failed)
                 {
-                    _logger.LogError(_failException, "{FailMessage} ({Duration}ms) [Timing {TimingId}]", _failMessage, Stopwatch.ElapsedMilliseconds, _timingId);
+                    if (_failMessageArgs == null)
+                    {
+                        _logger.LogError(_failException, "{FailMessage} ({Duration}ms) [Timing {TimingId}]", _failMessage, Stopwatch.ElapsedMilliseconds, _timingId);
+                    }
+                    else
+                    {
+                        var args = new object[_failMessageArgs.Length + 2];
+                        _failMessageArgs.CopyTo(args, 0);
+                        args[_failMessageArgs.Length - 1] = Stopwatch.ElapsedMilliseconds;
+                        args[_failMessageArgs.Length] = _timingId;
+                        _logger.LogError(_failException, _failMessage + " ({Duration}ms) [Timing {TimingId}]", args);
+                    }
                 }
-                else switch (_level)
+                else
                 {
-                    case LogLevel.Debug:
-                        _logger.LogDebug("{EndMessage} ({Duration}ms) [Timing {TimingId}]", _endMessage, Stopwatch.ElapsedMilliseconds, _timingId);
-                        break;
-                    case LogLevel.Information:
-                        _logger.LogInformation("{EndMessage} ({Duration}ms) [Timing {TimingId}]", _endMessage, Stopwatch.ElapsedMilliseconds, _timingId);
-                        break;
-                    // filtered in the ctor
-                    //default:
-                    //    throw new Exception();
+                    switch (_level)
+                    {
+                        case LogLevel.Debug:
+                            if (_endMessageArgs == null)
+                            {
+                                _logger.LogDebug("{EndMessage} ({Duration}ms) [Timing {TimingId}]", _endMessage, Stopwatch.ElapsedMilliseconds, _timingId);
+                            }
+                            else
+                            {
+                                var args = new object[_endMessageArgs.Length + 2];
+                                _endMessageArgs.CopyTo(args, 0);
+                                args[_endMessageArgs.Length - 1] = Stopwatch.ElapsedMilliseconds;
+                                args[_endMessageArgs.Length] = _timingId;
```

Comment:
While doing some testing I noticed that the log output was unexpected, shouldn't it be `args[args.Length - 1]` and not `args[_endMessageArgs.Length - 1]`?

For instance I provided one argument, so `Stopwatch.ElapsedMilliseconds` became the parameter in my message, and the `_timingId` became the `Duration` and `TimingId` was set to null.